### ISSUE TITLE
Multi condition if statements can produce some very wierd behaviour

### ISSUE THREAD

**Issue by taksuyu** [fish-shell/fish-shell#1583]
```
function check_path
    if test -d $argv[1]
        echo "$argv[1] made it through test"
        if not contains $argv[1] $PATH
            echo "$argv[1] will be added"
            set PATH $argv[1] $PATH
        end
    end
end
```
Works, but
```
function check_path
    if test -d $argv[1]; and not contains $argv[1] $PATH
            set PATH $argv[1] $PATH
    end
end
```
Doesn't

**Comment by faho** [comment]
See also #1428.

**Comment by faho** [comment]
This should be fixed like #1428.
